### PR TITLE
Homework 18

### PR DESCRIPTION
**Pull request by randyvergil**

### CODE REVIEW

File: src/test/java/Homework18.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class Homework18 extends BaseTest {
+
+    @Test
+
+    public void playSong() {
+      provideEmail("randy.davila@testpro.io");
```
Comment (khanhtestpro) [inline src/test/java/Homework18.java:12]: Hi,
I have checked your homework 18, but Test script is failed! You got this problem, please correct it.
Best wishes from the HW team (Khanh, khanh@testpro.io)

image


I don't see you add step navigate to koel web, you can add step `navigateToPage();` before getting into work  provideEmail

File: src/test/resources/TestNG.xml [hunk 2]
```diff
@@ -6,7 +6,12 @@
     <test name="Example Test" preserve-order="false">
         <classes>
             <class name="LoginTests"/>
+            <class name="ProfileTests"/>
             <class name="BaseTest"/>
+            <class name="Homework17"/>
+            <class name="Homework16"/>
+            <class name="Homework18"/>
```
Comment (khanhtestpro) [inline src/test/resources/TestNG.xml]: //comment from hw team: Homework requirement 18: XML File Configurations requires you to Use the <methods> & <include> tags in the XML file, but currently I have not seen you use it.

TestNg provides an option to include or exclude Groups, Test Methods, Classes and Packages using include and exclude tags by defining in testng.xml.
You will create a Class with Test Methods. and include test methods into xml file. in your Homewok18 you already have the playSong method, so you need to include that method into xml file

//HINT:
image

**Comment by randyvergil** [comment]
Attempt #4

**Comment by khanhtestpro** [comment]
Hi,
Your homework 18 has been reviewed again, but test suite TestNG.xml contains Homework18 still failed. It's got problem, plase check and correct it.
image

File: src/test/resources/TestNG.xml [hunk 3]
```diff
@@ -5,8 +5,23 @@
 
     <test name="Example Test" preserve-order="false">
         <classes>
+<<<<<<< Updated upstream
```
Comment (khanhtestpro) [inline src/test/resources/TestNG.xml]: please remove it

File: src/test/resources/TestNG.xml [hunk 4]
```diff
@@ -5,8 +5,23 @@
...
             <class name="LoginTests"/>
+            <class name="ProfileTests"/>
             <class name="BaseTest"/>
+            <class name="Homework17"/>
+            <class name="Homework16"/>
+            <class name="Homework18"/>
+=======
```
Comment (khanhtestpro) [inline src/test/resources/TestNG.xml]: the same, please remove it

File: src/test/resources/TestNG.xml [hunk 5]
```diff
@@ -5,8 +5,23 @@
...
+
+          <class name="Homework18">
+              <methods>
+                  <include name="playSong" />
+              </methods>
+
+          </class>
+>>>>>>> Stashed changes
```
Comment (khanhtestpro) [inline src/test/resources/TestNG.xml]: please remove it

File: src/test/java/Homework18.java (hunk 1 above)
Comment (khanhtestpro) [inline src/test/java/Homework18.java:12]: please add step `navigateToPage()` before step  `provideEmail("randy.davila@testpro.io");` as my previous comment

image

**Comment by randyvergil** [comment]
I have uploaded a new update called Attempt 3.1

**Pull request closed without merging**